randomerz/Slider
# Chad Dialogue and Item Drops (mostly fixed)

## Description
Changes dialogue system to use SaveSystem, Chad now properly drops Slider 6 and Boots after interacting with him when the player wins the race, adds a null check to STile.cs to fix a bug where Chad race was unwinnable (was impossible to interact/run into the bell)

## Related Task
chad race cheating player feedback systems so you know if youre racing or not

## How Has This Been Tested?
Run the jungle scene in Assets/Scenes/Dev/Avery/Jungle, grab Slider 2&3, run ES(5) in debug mode, then run to Chad. Set up the track to the bell.

To test PlayerLoses:
- Press E to start the race
- Intentionally lose to Chad
- Run to Chad near the bell
- Read his dialogue and press E when ready to race again

To test PlayerCheated:
- Press E to start the race
- Move Slider 2&3 or Slider 5 - some way that the track is broken
- Observe Chad's dialogue when the track is broken
- Realign the track the way it was before to fix it
- Observe Chad's dialogue when the track is fixed
- Press E when ready to race again

To test PlayerWon:
- Press E to start the race
- Win the race and wait for Chad to catch up
- Observe the dialogue and press E to progress the dialogue until Chad drops the Boots and Slider 6
- Pick up Slider 6 and observe Chad's dialogue change

For all testing:
- Check Debug logs/warnings/errors for any additional issues

## Screenshots (if appropriate):
